tests/unittests/test_net.py

```diff
@@ -8234,10 +8234,11 @@ def test_duplicate_ignored_macs(self):
         self._mock_setup()
         with does_not_raise():
             net.get_interfaces_by_mac()
-        assert (
-            "Ignoring duplicate macs from 'swp0' and 'swp1' due to "
+        pattern = (
+            "Ignoring duplicate macs from 'swp[0-1]' and 'swp[0-1]' due to "
```

Comment:
@TheRealFalcon I think this is python-version-specific for how `list`s are generated from `set`s?  Given that _mock_setup is creating a set from a list of mac,keys() without explicitly calling sorted, I think that's what we are hitting here.

Let's try this diff instead.
```diff
diff --git a/tests/unittests/test_net.py b/tests/unittests/test_net.py
index 056aaeb6f..2e58e685e 100644
--- a/tests/unittests/test_net.py
+++ b/tests/unittests/test_net.py
@@ -7862,7 +7862,7 @@ class TestGetInterfaces(CiTestCase):
     data: dict = {}
 
     def _se_get_devicelist(self):
-        return list(self.data["devices"])
+        return sorted(list(self.data["devices"]))
 
     def _se_device_driver(self, name):
         return self.data["drivers"][name]
@@ -8234,11 +8234,10 @@ class TestGetInterfacesByMac(CiTestCase):
         self._mock_setup()
         with does_not_raise():
             net.get_interfaces_by_mac()
-        pattern = (
-            "Ignoring duplicate macs from 'swp[0-1]' and 'swp[0-1]' due to "
+        assert (
+            "Ignoring duplicate macs from 'swp0' and 'swp1' due to "
             "driver 'mscc_felix'."
-        )
-        assert re.search(pattern, self.logs.getvalue())
+        ) in self.logs.getvalue()
 
 
 class TestInterfacesSorting(CiTestCase):
```

Reply:
No such luck. py 3.7 then failed. :/